### ISSUE TITLE
Rule for inputs without an associated label

### ISSUE THREAD

**Issue by badtant** [evcohen/eslint-plugin-jsx-a11y#66]
Is there a rule for making sure every input has an associated label? I've switched on the rule "label-has-for" but it only checks whether the label has a for attribute. Lets say I just have an input and no label to it, that should produce an error.

``` jsx
<input type="text" id="id1" />
```

VS

``` jsx
<label for="id1">Some text</label>
<input type="text" id="id1" />
```

Also when a label is empty, there should be an error.

**Comment by ljharb** [comment]
Ideally it also requires wrapping the input in the label, with an option to allow it to not be wrapped. (For full a11y coverage, a label should both wrap it's input and be for/id-linked)

**Comment by evcohen** [comment]
I think both can be enforced if both the DOM elements or the wrapper components (as soon as global config from #65 is implemented) are in the same scope. Also should pass if there is [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) linkage

**Comment by jessebeach** [comment]
You're in luck @badtant, I've got a PR with a rule for the behavior that you describe.

#326
